Simbody/include/simbody/internal/CableSpan.h

```diff
@@ -0,0 +1,358 @@
+
+    /** Get the derivative of the total cable length.
+     * State must be realized to Stage::Position. */
+    Real getLengthDot(const State& state) const;
+
+    void applyBodyForces(
+        const State& state,
+        Real tension,
+        Vector_<SpatialVec>& bodyForcesInG) const;
+
+    /** Compute the cable power.
+     * State must be realized to Stage::Position. */
+    Real calcCablePower(const State& state, Real tension) const;
+
+    /** Compute points on the path spanned by this cable.
+     * The curve segments will be resampled from the underlying geodesic such
+     * that the computed points are at equal length intervals. The straight
+     * line segments are not resampled.
+     * State must be realized to Stage::Position.
+     *
+     * @param state System State.
+     * @param lengthInterval Each curve segments will be resampled such the the
+     * points interspacing does not exceed this value.
+     * @param sink Where the path points will be written to.
+     * @return The total number of computed points.
+     */
+    int calcPathPoints(
+        const State& state,
+        Real lengthInterval,
+        std::function<void(Real length, Vec3 point)> sink) const;
+
+    class Impl;
+
+private:
+    /** Wrap the Impl, increasing the reference count. */
+    explicit CableSpan(std::shared_ptr<Impl> impl): m_Impl(std::move(impl)) {}
+
+    /** Cheap copy of pointer to the Impl. */
+    CableSpan copyImpl() const {
+        return CableSpan(m_Impl);
+    }
+
+    const Impl& getImpl() const
+    {
+        return *m_Impl;
+    }
+
+    Impl& updImpl()
+    {
+        return *m_Impl;
+    }
+
+    std::shared_ptr<Impl> m_Impl = nullptr;
+
+    friend CableSubsystem;
+
+    // Befriend the helper class for testing the implementation.
+    friend CableSubsystemTestHelper;
+};
+
+//==============================================================================
+//                                CABLE SUBSYSTEM
+//==============================================================================
+
+class SimTK_SIMBODY_EXPORT CableSubsystem : public Subsystem
+{
+public:
+    CableSubsystem();
+    explicit CableSubsystem(MultibodySystem&);
+
+    int getNumCables() const;
+    const CableSpan& getCable(CableSpanIndex idx) const;
+    CableSpan& updCable(CableSpanIndex idx);
+
+    SimTK_PIMPL_DOWNCAST(CableSubsystem, Subsystem);
+    class Impl;
+    Impl& updImpl();
+    const Impl& getImpl() const;
+
+    // Befriend the helper class for testing the implementation.
+    friend CableSubsystemTestHelper;
+};
+
+//==============================================================================
+//                      SUBSYSTEM TESTING HELPER
+//==============================================================================
+
+// Helper class for testing the internally computed path error jacboian.
+class SimTK_SIMBODY_EXPORT CableSubsystemTestHelper
+{
+public:
+    CableSubsystemTestHelper() = default;
```

Comment:
Probably unnecessary: you only need to default a zero-arg ctor if a nonzero one exists